pkg/tfbridge/schema.go

```diff
@@ -1652,6 +1652,33 @@ func isDefaultOrZeroValue(tfs shim.Schema, ps *SchemaInfo, v resource.PropertyVa
 func extractSchemaInputs(state resource.PropertyValue, tfs shim.Schema, ps *SchemaInfo,
 	rawNames bool) resource.PropertyValue {
 
+	if ps == nil {
+		ps = &SchemaInfo{}
+	}
+
+	// If ps is MaxItemsOne, then it will not be present in the
+	for IsMaxItemsOne(tfs, ps) {
+		ps = ps.Elem
+		if ps == nil {
+			ps = &SchemaInfo{}
+		}
+
+		switch e := tfs.Elem().(type) {
+		case shim.Schema:
+			tfs = e
+		case shim.Resource:
+			if state.IsObject() {
+				return resource.NewProperty(
```

Comment:
should this be NewObjectProperty to be more type-precise (nit)

Reply:
Can you clarify what you mean by "more type-precise". `NewObjectProperty` is more explicit, but I don't see why we shouldn't use type inference. [`NewProperty`](https://github.com/pulumi/pulumi/blob/28f098a660aa698c83a77058da28f7867707d0a0/sdk/go/common/resource/properties.go#L237-L240) is fully type safe:

```go
// PropertyValueType enumerates the actual types that may be stored in a PropertyValue.
//
//nolint:lll
type PropertyValueType interface {
	bool | float64 | string | *asset.Asset | *archive.Archive | Computed | Output | *Secret | ResourceReference | []PropertyValue | PropertyMap
}

// NewProperty creates a new PropertyValue.
func NewProperty[T PropertyValueType](v T) PropertyValue {
	return PropertyValue{v}
}
```

Reply:
Just out of habit, I'd use least general primitives, in case the underlying type switches to string, that's a bug. But it's purely theoretical so disregard the nit.